topics/list_topics.go

```diff
@@ -0,0 +1,35 @@
+package topics
+
+import (
+	"context"
+	"regexp"
+
+	kafkago "github.com/segmentio/kafka-go"
+)
+
+// ListTopics returns a slice of all the Topics
+func ListTopics(ctx context.Context, client *kafkago.Client) (topics []kafkago.Topic, err error) {
+	response, err := client.Metadata(ctx, &kafkago.MetadataRequest{
+		Addr: client.Addr,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return response.Topics, nil
+}
+
+// ListTopicsRegex returns a slice of Topics that match a regex
+func ListTopicsRegex(ctx context.Context, client *kafkago.Client, match regexp.Regexp) (topics []string, err error) {
```

Comment:
Thoughts on adding a test for this?

Reply:
I had one with my initial implementation but the testing framework isn't available as this is a new package. I also tried making a mock but there are not interfaces to mock. If you are OK with me adding an interface or two as necessary, I'll go that direction.

Reply:
Hmm yeah it's not a great setup for this as is, if it's not too much extra work we could try moving functions as needed to the `testing` package? https://github.com/segmentio/kafka-go/tree/master/testing

Reply:
I'd rather not extend this PR to include rewriting the test framework.

Reply:
If you would rather not copy functions into the `testing` package, what do you think about copying the functions you need into the `list_topics_test.go` file itself?

Reply:
Done